kasa/smart/smartmodule.py

```diff
@@ -76,9 +76,13 @@ def __init__(self, device: SmartDevice, module: str):
         self._error_count = 0
 
     def __init_subclass__(cls, **kwargs):
-        name = getattr(cls, "NAME", cls.__name__)
-        _LOGGER.debug("Registering %s", cls)
-        cls.REGISTERED_MODULES[name] = cls
+        parent_name = ".".join(cls.__module__.split(".")[:-1])
+        expected_name = ".".join(__name__.split(".")[:-1]) + ".modules"
+        # only register subclasses in the relative .modules package
```

Comment:
This feels rather hacky and it's not directly clear why this is done. Perhaps there's some better way to achieve the same behavior? Is this something that's just needed because we have experimental, would scrapping it in favor of cleaner code make sense?

Reply:
I've simplified it in https://github.com/python-kasa/python-kasa/pull/1189 and added a comment.